src/openrct2/scripting/ScEntity.hpp

```diff
@@ -119,6 +119,40 @@ namespace OpenRCT2::Scripting
             }
         }
 
+        void remove()
+        {
+            auto ctx = GetContext()->GetScriptEngine().GetContext();
+            auto entity = GetEntity();
+            if (entity != nullptr)
+            {
+                switch (entity->sprite_identifier)
+                {
+                    case SPRITE_IDENTIFIER_VEHICLE:
+                        duk_error(ctx, DUK_ERR_ERROR, "Removing a vehicle is currently unsupported.");
+                        break;
+                    case SPRITE_IDENTIFIER_PEEP:
+                    {
+                        auto peep = static_cast<Peep*>(entity);
+                        // We can't remove a single peep from a ride at the moment as this can cause complications with the
+                        // vehicle car having an unsupported peep capacity.
+                        if (peep->state == PEEP_STATE_ON_RIDE || peep->state == PEEP_STATE_ENTERING_RIDE)
+                        {
+                            duk_error(ctx, DUK_ERR_ERROR, "Removing a peep that is on a ride is currently unsupported.");
+                        }
+                        else
+                        {
+                            peep->Remove();
+                        }
+                        break;
+                    }
+                    default:
+                        invalidate_sprite_2(entity);
+                        sprite_remove(entity);
+                        break;
```

Comment:
Do we want a `default` branch? There are only three other options, I tend to not like handling defaults because if something is added later it will automatically handle (and maybe incorrectly)